packages/lib/src/toolbar/controls/Selector/Selector.tsx

```diff
@@ -28,54 +37,136 @@ function Selector<T extends string>(props: Props<T>) {
+    [
+      useClick(context),
+      useFloatingDismiss(context),
+      useListNavigation(context, {
+        listRef,
+        activeIndex,
+        loop: true,
+        focusItemOnHover: false,
+        onNavigate: setActiveIndex,
+      }),
+    ],
+  );
+
+  function handleSelect(index: number, option: T) {
+    setSelectedIndex(index);
+    toggle(false);
+    onChange(option);
+  }
 
   return (
     <div className={styles.root}>
-      {label && <span className={styles.selectorLabel}>{label}</span>}
+      {label && (
+        <span id={labelId} className={styles.selectorLabel}>
+          {label}
+        </span>
+      )}
 
-      <Wrapper className={styles.wrapper} onSelection={onChange}>
-        <Button className={styles.btn} tag="button" disabled={disabled}>
-          <div className={styles.btnLike}>
-            <span className={styles.selectedOption}>
-              <Option option={value} isSelected />
-            </span>
-            <MdArrowDropDown className={styles.arrowIcon} />
-          </div>
-        </Button>
+      <button
+        ref={refs.setReference}
+        id={referenceId}
+        className={styles.btn}
+        type="button"
+        disabled={disabled}
+        role="combobox"
+        aria-labelledby={`${label ? labelId : ''} ${selectedOptionId}`}
+        aria-haspopup="listbox"
+        aria-expanded={isOpen || undefined}
+        aria-controls={context.floatingId}
+        {...getReferenceProps()}
+      >
+        <span className={styles.btnLike}>
+          <span id={selectedOptionId} className={styles.selectedOption}>
+            <OptionComp option={value} isSelected />
+          </span>
+          <MdArrowDropDown className={styles.arrowIcon} />
+        </span>
+      </button>
 
-        <Menu
+      {isOpen && (
+        <div
+          ref={refs.setFloating}
+          id={context.floatingId}
           className={styles.menu}
-          style={{ maxHeight: Math.min(MENU_IDEAL_HEIGHT, menuMaxHeight) }}
+          style={floatingStyles}
+          role="listbox"
+          aria-labelledby={referenceId}
+          {...getFloatingProps()}
         >
-          {Array.isArray(options) ? (
-            <OptionList
-              optionList={options}
-              optionComponent={Option}
-              value={value}
-            />
-          ) : (
-            <ul className={styles.list}>
-              {Object.entries(options).map(([groupLabel, groupOptions]) => (
-                <li key={groupLabel}>
-                  <span className={styles.groupLabel}>{groupLabel}</span>
-                  <ul className={styles.list}>
-                    <OptionList
-                      optionList={groupOptions}
-                      optionComponent={Option}
-                      value={value}
-                    />
-                  </ul>
-                </li>
-              ))}
-            </ul>
-          )}
-        </Menu>
-      </Wrapper>
+          <FloatingList elementsRef={listRef}>
```

Comment:
This is the `FloatingList` I was talking about. It's basically just a context under the hood.